webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -289,23 +290,16 @@ trait PromiseGenerationService {
                                                             allRules             <- findDependantRules() ?~! "Could not find dependant rules"
                                                             fetch1Time            = System.currentTimeMillis
                                                             _                     = PolicyGenerationLogger.timing.trace(s"Fetched rules in ${fetch1Time - fetch0Time} ms")
-                                                            nodeFacts            <-
-                                                              getNodeFacts().map(_.filter {
-                                                                case (_, n) =>
-                                                                  if (n.rudderSettings.state == NodeState.Ignored) {
-                                                                    PolicyGenerationLogger.debug(
-                                                                      s"Skipping node '${n.id.value}' because the node is in state '${n.rudderSettings.state.name}'"
-                                                                    )
-                                                                    false
-                                                                  } else true
-                                                              }) ?~! "Could not get Node Infos" // disabled node don't get new policies
+                                                            nf                   <- getNodeFacts() ?~! "Could not get Node Infos" // disabled node don't get new policies
                                                             fetch2Time            = System.currentTimeMillis
                                                             _                     = PolicyGenerationLogger.timing.trace(s"Fetched node infos in ${fetch2Time - fetch1Time} ms")
                                                             directiveLib         <-
                                                               getDirectiveLibrary(allRules.flatMap(_.directiveIds).toSet) ?~! "Could not get the directive library"
                                                             fetch3Time            = System.currentTimeMillis
                                                             _                     = PolicyGenerationLogger.timing.trace(s"Fetched directives in ${fetch3Time - fetch2Time} ms")
-                                                            groupLib             <- getGroupLibrary() ?~! "Could not get the group library"
+                                                            gl                   <- getGroupLibrary() ?~! "Could not get the group library"
+                                                            tmp                  <- GetConsistentNodesAndGroups(nf, gl).toBox
+                                                            (nodeFacts, groupLib) = tmp
```

Comment:
This is the main change. Here we have a new `groupLib`, consistent with `nodeFacts` actual list.